MinGW, CMake and project layout

squimrel opened the pull request:
I'm pretty unsure about the MinGW package name in other words this needs review.

bcl commented:
I don't have any interesting in making this MinGW compatible. The changes are too extensive for something that won't be used (tested) by the majority of the users so it'll end up rotting.

squimrel commented:
I need this for the [Fedora Media Writer](https://github.com/MartinBriza/MediaWriter/) because it should work on Windows. I don't know who else is a user of isomd5sum apart from people using the command-line tool I can't find any projects that use it.

I could have also changed the Makefile but doing it properly using CMake seemed cleaner in the long run.
Also it's supposed to make packaging easier and especially take away the work from you to package for mingw that's why there're so many changes.

bcl commented:
Why does media writer need it? Is it going to be doing a pre-check of the embedded checksums instead of (or in addition to) the SHA of the iso?

squimrel commented:
No it does the SHA-256 check after download. But to ensure that nothing went wrong after writing to disk it's doing a isomd5sum check and then after it modified the partition table it updates the isomd5sum. So it needs both check and implant.

squimrel commented:
I forgot to mention it also implants re-implants the checksum after modifying the iso image.

bcl commented:
> Then it might modify the iso image and therefore re-implant the checksum.

Ah, ok! Thanks for the explanation.

squimrel commented:
> I don't have any interesting in making this MinGW compatible. The changes are too extensive for something that won't be used (tested) by the majority of the users so it'll end up rotting.

Since you didn't close it you must be still thinking about it. Any thoughts you'd like to share?

bcl commented on src/checkisomd5.c:
```diff
@@ -24,8 +24,8 @@
 
-#include "md5.h"
-#include "libcheckisomd5.h"
+#include "./include/libcheckisomd5.h"
```
This shouldn't be needed if you can pass the include path to it from cmake

squimrel replied:
Well do I want to tell cmake to search in include?

bcl replied:
Yes, telling cmake about ./include/ is the right way to do it.

squimrel commented:
@bcl Okay in that case I'll remove all the packaging stuff and leave that to the packager.

bcl commented:
I think there's a couple other things to add/figure out:
* version number -- you have it pulling from the git tag, but that doesn't help if someone has a bare archive of the code. It used to be in the Makefile and tagged from that.
* Creating an archive -- There should be a make target for this like there used to be: https://github.com/rhinstaller/isomd5sum/blob/master/Makefile#L72
* Add ./build/ to .gitignore
* Add a section to the README.md on how to create a new release.

Thanks!

squimrel commented:
Sorry for the late response. I addressed your last comment. But AFAIK you can not add custom rules to the Makefile that cmake creates so instead I added a script which creates that archive at `scripts/create-archive`.

bcl commented:
Thanks for the updated patches, but I'm going to have to nak these changes. I've spend the afternoon trying to get it working with an updated fedora spec file and I keep hitting problems. It just introduces too much change for my taste.
